packages/automerge-repo/src/DocHandle.ts

```diff
@@ -78,156 +63,93 @@ export class DocHandle<T> //
 
     this.#log = debug(`automerge-repo:dochandle:${this.documentId.slice(0, 5)}`)
 
-    /**
-     * Internally we use a state machine to orchestrate document loading and/or syncing, in order to
-     * avoid requesting data we already have, or surfacing intermediate values to the consumer.
-     *
-     *                          ┌─────────────────────┬─────────TIMEOUT────►┌─────────────┐
-     *                      ┌───┴─────┐           ┌───┴────────┐            │ unavailable │
-     *  ┌───────┐  ┌──FIND──┤ loading ├─REQUEST──►│ requesting ├─UPDATE──┐  └─────────────┘
-     *  │ idle  ├──┤        └───┬─────┘           └────────────┘         │
-     *  └───────┘  │            │                                        └─►┌────────┐
-     *             │            └───────LOAD───────────────────────────────►│ ready  │
-     *             └──CREATE───────────────────────────────────────────────►└────────┘
```

Comment:
I've added a link to a visualization that's generated from the code; I have to copy-paste the code into that when it changes, but it will still be easier to keep up to date. 

We can also use the XState extension for VS Code to see the state chart, generated directly from the code.